### PR TITLE
Created a couple of preparatory classes

### PR DESCRIPTION
**Pull request by oldalton**
* A couple of preparatory classes necessary for token cache (token response, token request)
* Moved a couple of additional classes from ADAL/MSAL (json object, client info)
* Moved accompanying tests
* Added a generic token cache interface to be implemented by ADAL and MSAL SDKs

### CODE REVIEW

File: IdentityCore/src/oauth2/aad/MSIDClientInfo.h [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+#import "MSIDJsonObject.h"
+
+@interface MSIDClientInfo : MSIDJsonObject
```
Comment (unpluggedk) [inline IdentityCore/src/oauth2/aad/MSIDClientInfo.h]: This looks very much like our MSIDUser class although they have 2 distinct needs. 
Should we combine them?

File: IdentityCore/src/util/MSIDJsonObject.h [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+#define MSID_JSON_ACCESSOR(KEY, GETTER) DICTIONARY_READ_PROPERTY_IMPL(_json, KEY, GETTER)
+#define MSID_JSON_MUTATOR(KEY, SETTER) DICTIONARY_WRITE_PROPERTY_IMPL(_json, KEY, SETTER)
+
+#define MSID_JSON_RW(KEY, GETTER, SETTER) \
+    MSID_JSON_ACCESSOR(KEY, GETTER) \
+    MSID_JSON_MUTATOR(KEY, SETTER)
+
+@interface MSIDJsonObject : NSObject
+{
+    NSMutableDictionary *_json;
```
Comment (unpluggedk) [inline IdentityCore/src/util/MSIDJsonObject.h]: nit: Does this need to be mutable?

Comment (oldalton) [reply IdentityCore/src/util/MSIDJsonObject.h]: I think yes, if any properties need to be modified afterwards

File: IdentityCore/src/oauth2/aad/MSIDClientInfo.h (hunk 1 above)
Comment (oldalton) [reply IdentityCore/src/oauth2/aad/MSIDClientInfo.h]: This is just the representation of what we got back from the server. MSIDUser has a much wider purpose and not sure if we should combine it.

**Pull request merged**